extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java

```diff
@@ -230,4 +231,19 @@ public String makeIndexPathName(DataSegment dataSegment, String indexName)
         indexName
     );
   }
+
+  private void initFullyQualifiedStorageDirectory()
```

Comment:
Would you please add a comment about why this should be lazily initialized?

Reply:
Whoops, forgot to push that change, added a comment